samples/openthread/ncp/prj.conf

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+# ===============================
+# KERNEL OPTIONS
+# ===============================
+CONFIG_NEWLIB_LIBC=y
+CONFIG_MAIN_STACK_SIZE=4096
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_INIT_STACKS=y
+# ===============================
+
+# ===============================
+# LOGGING OPTIONS
+# ===============================
+CONFIG_NET_LOG=y
+CONFIG_LOG=y
+CONFIG_NET_STATISTICS=y
+CONFIG_PRINTK=y
+CONFIG_OT_NCP_LOG_LEVEL_DBG=y
+CONFIG_NET_CONFIG_LOG_LEVEL_OFF=y
+
+CONFIG_LOG_BACKEND_SPINEL=y
+CONFIG_LOG_BACKEND_RTT=n
+CONFIG_LOG_BACKEND_UART=y
+CONFIG_BOOT_BANNER=n
+CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE=2048
+CONFIG_LOG_BACKEND_SPINEL_LEVEL_INFO=y
+# ===============================
+
+# ===============================
+# NETWORKING OPTIONS
+# ===============================
+# ====================
+# OPENTHREAD OPTIONS
+# ====================
+# set OpenThread NCP architecture
+CONFIG_OPENTHREAD_NCP=y
+
+# enable OpenThread
+CONFIG_NET_L2_OPENTHREAD=y
+
+CONFIG_OPENTHREAD_DEBUG=y
+CONFIG_OPENTHREAD_L2_DEBUG=y
+CONFIG_OPENTHREAD_L2_LOG_LEVEL_INF=y
+CONFIG_OPENTHREAD_CHANNEL=11
+# Use PANID 0xffff to be able to commission
+CONFIG_OPENTHREAD_PANID=65535
+
+# stack sizes configuration
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=5120
+CONFIG_SHELL_STACK_SIZE=3072
+CONFIG_OPENTHREAD_THREAD_STACK_SIZE=8192
+CONFIG_OPENTHREAD_PKT_LIST_SIZE=20
+
+# Other OpenThread features
+CONFIG_OPENTHREAD_DHCP6_SERVER=y
+CONFIG_OPENTHREAD_COMMISSIONER=y
+CONFIG_OPENTHREAD_BORDER_AGENT=y
+CONFIG_OPENTHREAD_BORDER_ROUTER=y
+CONFIG_OPENTHREAD_UDP_FORWARD=y
+CONFIG_OPENTHREAD_ENABLE_SERVICE=y
+
+# ====================
+
+# ====================
+# COMMON OPTIONS
+# ====================
+# Generic
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_CONFIG_NEED_IPV6=y
+
+# disable certain parts of Zephyr IPv6 stack
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+
+CONFIG_NET_TCP=n
+CONFIG_NET_IPV4=n
+CONFIG_NET_CONFIG_NEED_IPV4=n
+
+# make sure Bluetooth stack is disabled
+CONFIG_BT=n
+
+# reduce very wide OpenThread addressing register
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=6
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=8
+CONFIG_NET_MAX_CONTEXTS=10
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=16
```

Comment:
not sure if those are needed? do other samples have all of those?

Reply:
COAP samples have such settings, Zephyr sample either, so I thought it is somehow needed.